src/MatrixNormalForms.jl

```diff
@@ -0,0 +1,73 @@
+################################################################################
+#
+#  Matrix normal form over fields (echelon_form)
+#
+################################################################################
+
+@doc raw"""
+    echelon_form(A::MatElem{<:FieldElement}; reduced::Bool = true, shape::Symbol = :upper, trim::Bool = false)
+
+Return a row echelon form $R$ of $A$.
+
+# Keyword arguments
+* `reduced`: Whether the columns of $R$ which contain a pivot are reduced. The
+  default is `true`.
+* `shape`: Whether $R$ is an upper-right (`:upper`, default) or lower-left
+  (`:lower`) triangular matrix.
```

Comment:
Is this correct? Or do we want more/other triangular shapes?

Reply:
These two should be sufficient. But I learned the other day, that `triangular matrix` means something different. Maybe replace it by `echelon form`.